### PR TITLE
marsRover17: Sends form data to /authenticate

### PR DESCRIPTION
**Pull request by HeatherJLewis**
**Description**
Create an authenticate endpoint which receives data from the html login form.

**Changes**
* adds body-parser package
* adds `/authenticate` endpoint
* wraps the input elements in a `<form>` tag
* adds `express.urlencoded()` middle ware to extract the form data

### CODE REVIEW

File: frontend/login.html [hunk 1]
```diff
@@ -12,19 +12,21 @@
...
+                    <input type="text" id="username" name="username">
+                </div>
+                <div class="form-input">
+                    <label for="password">Password</label>
+                    <input type="password" id="password" name="password">
+                </div>
+                <button type="submit">
+                    Submit
+                </button>
             </div>
```
Comment (Kenneth-Siu) [inline frontend/login.html:19]: For this, do we strictly need this div?

**Review by Kenneth-Siu — COMMENTED** [review]
Just the one thought :)

File: frontend/login.html (hunk 1 above)
Comment (HeatherJLewis) [reply frontend/login.html:19]: No I'm. not sure that we do. I think the `<form>` tags would supercede the `<div>`. I'll change it. :)

Comment (HeatherJLewis) [reply frontend/login.html:19]: Done :)

**Review by Kenneth-Siu — APPROVED** [review]

**Pull request merged**